### DISCUSSION TITLE
Rocket fails to comprehend MultiIndex dimensions, multiplies actual value count with the length of the first index

### DISCUSSION THREAD

**Discussion by MagnusNordboe** [alan-turing-institute/sktime#4293] [Q&A]
After rewriting my code per https://github.com/sktime/sktime/discussions/4260 I've come to the point of actually training on it.

I have a MultiIndex DataFrame representing sets of MTS data. Currently it is of shape (3,399,586). Representing 3 different instances of the MTS data, with 399 rows and 586 columns. 

What happens here is quite clear: It takes an input dataframe of shape (399,586), that has  233814 values. (corresponding to 399*586). Then it tries to paste them into a shape of (3,399,586), expecting a value size three times the actual value size. It seems to not comprehend the MultiIndex format.

Any quick fix for this?

Here's a zip file with the files and python code:

[problem_demonstration.zip](https://github.com/sktime/sktime/files/10898832/problem_demonstration.zip)

```
file_list = ["./catalog.csv",
             "./catalog2.csv",
             "./even_load.csv"]

complete_df = readcsvs(file_list)
print(complete_df.shape)
print(complete_df.values.size)
```

Output: 

```
(399, 586)

233814
```

Then I try to run Rocket on this data:

```
from sktime.transformations.panel.rocket import Rocket

rocket = Rocket()
rocket.fit(complete_df)
```

Output:

```
Output exceeds the [size limit](command:workbench.action.openSettings?%5B%22notebook.output.textLineLimit%22%5D). Open the full output data [in a text editor](command:workbench.action.openLargeOutput?5f9f37bb-4aa4-4050-958f-b016eb997fad)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
[~\AppData\Local\Temp\ipykernel_7300\3325979802.py](https://file+.vscode-resource.vscode-cdn.net/f%3A/Master/Kubernetes/sockshop/microservices-demo/analysis/problem_demonstration/~/AppData/Local/Temp/ipykernel_7300/3325979802.py) in 
      2 
      3 rocket = Rocket()
----> 4 rocket.fit(complete_df)

[f:\anaconda\envs\py39\lib\site-packages\sktime\transformations\base.py](file:///F:/anaconda/envs/py39/lib/site-packages/sktime/transformations/base.py) in fit(self, X, y)
    396 
    397         # check and convert X/y
--> 398         X_inner, y_inner = self._check_X_y(X=X, y=y)
    399 
    400         # memorize X as self._X, if remember_data tag is set to True

[f:\anaconda\envs\py39\lib\site-packages\sktime\transformations\base.py](file:///F:/anaconda/envs/py39/lib/site-packages/sktime/transformations/base.py) in _check_X_y(self, X, y, return_metadata)
    940         if not requires_vectorization:
    941             # converts X
--> 942             X_inner = convert_to(
    943                 X,
    944                 to_type=X_inner_mtype,

[f:\anaconda\envs\py39\lib\site-packages\sktime\datatypes\_convert.py](file:///F:/anaconda/envs/py39/lib/site-packages/sktime/datatypes/_convert.py) in convert_to(obj, to_type, as_scitype, store, store_behaviour)
    261             to_type = same_scitype_mtypes[0]
    262 
--> 263     converted_obj = convert(
...
--> 584     X_3d = X.values.reshape(n_instances, n_timepoints, n_columns).swapaxes(1, 2)
    585 
    586     return X_3d

ValueError: cannot reshape array of size 233814 into shape (3,399,586)
```

**Comment by fkiraly** [comment]
@magnusnordboe, I cannot reproduce your error with dummy data.
Can you provide complete code, with real or dummy data?

For me, this works as expected:

```python
import numpy as np
from sktime.transformations.panel.rocket import Rocket

X = np.random.normal(size=(399, 586))
t = Rocket()
t.fit(X)
```

Perhaps you are using an old `sktime` version? What is your `sktime` version, python version, operating system?

**Reply by MagnusNordboe** [reply]
Hi Franz, thank you for reaching out again!

The zip file in my original post contains the training data as well as the code that transforms it from csv to a multiIndex DataFrame.

Here it is again, for clarity: 
[problem_demonstration.zip](https://github.com/sktime/sktime/files/10921387/problem_demonstration.zip)

I'm running sktime version 0.16.0, with python version 3.9.16, on Windows 10.

I've run Rocket on my machine with some other test data using MultiIndex, and it works fine. So something seems wrong with my indexing, but it looks to fulfill the spec sheet just fine: The inner index is a DateTimeIndex, outer is an object index denoting the time series. 

Quite stuck on this.
